wg-node-lifecycle/charter.md

```diff
@@ -0,0 +1,117 @@
+# WG Node Lifecycle Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
+the Roles and Organization Management outlined in [wg-governance].
+
+[Kubernetes Charter README]: /committee-steering/governance/README.md
+
+## Scope
+
+The Kubernetes ecosystem currently faces challenges in node maintenance scenarios, with multiple
+projects independently addressing similar issues. The goal of this working group is to develop
+unified APIs that the entire ecosystem can depend on, reducing the maintenance burden across
+projects and addressing scenarios that impede node drain or cause improper pod termination. Our
+objective is to create easily configurable, out-of-the-box solutions that seamlessly integrate with
+existing APIs and behaviors.
+
+To properly solve the node drain, we must first understand the node lifecycle. This includes
+provisioning/sunsetting of the nodes, PodDisruptionBudgets, API-initiated eviction and node
+shutdown. This then impacts both the node and pod autoscaling, load balancing, and the applications
+running in the cluster. All of these areas have issues and would benefit from a unified approach.
+
+### In scope
+
+- Explore a unified way of draining the nodes and managing node maintenance by introducing new APIs
```

Comment:
Can you please add a goal of migrating existing scenarios to the new API so the group will be tasked to not break users when they are upgrading

Reply:
We do have that in scope under

```
Migrate users of the eviction based kubectl-like drain (kubectl, cluster autoscaler, karpenter), and other scenarios to use the new approach.
```

So far it is pretty generic until we have a clearer vision. Please let me know if you would like to see something more specific.